core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -169,6 +182,31 @@ private[spark] class EventLoggingListener(
 
   // Events that trigger a flush
   override def onStageCompleted(event: SparkListenerStageCompleted): Unit = {
+    if (shouldLogExecutorMetricsUpdates) {
+      // clear out any previous attempts, that did not have a stage completed event
```

Comment:
Tracking task start and end would be some amount of overhead. If it's a relatively unlikely corner case, and unlikely to have much impact on the numbers, it may be better to leave as is.